floors/rule.go

```diff
@@ -0,0 +1,556 @@
+	}
+
+	if width != 0 && height != 0 {
+		size = fmt.Sprintf("%dx%d", width, height)
+	}
+	return size
+}
+
+// getBannerSize returns width and height for given banner impression
+func getBannerSize(imp *openrtb2.Imp) (int64, int64) {
+	width := int64(0)
+	height := int64(0)
+
+	if len(imp.Banner.Format) == 1 {
+		return imp.Banner.Format[0].W, imp.Banner.Format[0].H
+	} else if len(imp.Banner.Format) > 1 {
+		return width, height
+	} else if imp.Banner.W != nil && imp.Banner.H != nil {
+		width = *imp.Banner.W
+		height = *imp.Banner.H
+	}
+	return width, height
+}
+
+// getDomain returns domain provided into site or app object
+func getDomain(request *openrtb_ext.RequestWrapper) string {
+	var value string
+	if request.Site != nil {
+		if len(request.Site.Domain) > 0 {
+			value = request.Site.Domain
+		} else if request.Site.Publisher != nil && len(request.Site.Publisher.Domain) > 0 {
+			value = request.Site.Publisher.Domain
+		}
+	} else if request.App != nil {
+		if len(request.App.Domain) > 0 {
+			value = request.App.Domain
+		} else if request.App.Publisher != nil && len(request.App.Publisher.Domain) > 0 {
+			value = request.App.Publisher.Domain
+		}
+	}
+	return value
+}
+
+// getSiteDomain  returns domain provided into site object
+func getSiteDomain(request *openrtb_ext.RequestWrapper) string {
+	var value string
+	if request.Site != nil {
+		value = request.Site.Domain
+	} else {
+		value = request.App.Domain
+	}
+	return value
+}
+
+// getPublisherDomain returns publisher domain provided into site or app object
+func getPublisherDomain(request *openrtb_ext.RequestWrapper) string {
+	value := catchAll
+	if request.Site != nil && request.Site.Publisher != nil && len(request.Site.Publisher.Domain) > 0 {
+		value = request.Site.Publisher.Domain
+	} else if request.App != nil && request.App.Publisher != nil && len(request.App.Publisher.Domain) > 0 {
+		value = request.App.Publisher.Domain
+	}
+	return value
+}
+
+// getBundle returns app bundle type
+func getBundle(request *openrtb_ext.RequestWrapper) string {
+	value := catchAll
+	if request.App != nil && len(request.App.Bundle) > 0 {
+		value = request.App.Bundle
+	}
+	return value
+}
+
+// getGptSlot returns gptSlot
+func getGptSlot(imp *openrtb_ext.ImpWrapper) string {
+	value := catchAll
+
+	impExt, err := imp.GetImpExt()
+	if err == nil {
+		extData := impExt.GetData()
+		if extData != nil {
+			if extData.AdServer != nil && extData.AdServer.Name == "gam" {
+				gptSlot := extData.AdServer.AdSlot
+				if gptSlot != "" {
+					value = gptSlot
+				}
+			} else if extData.PbAdslot != "" {
+				value = extData.PbAdslot
+			}
+		}
+	}
+	return value
+}
+
+// getChannelName returns channel name
+func getChannelName(bidRequest *openrtb_ext.RequestWrapper) string {
+
+	reqExt, err := bidRequest.GetRequestExt()
+	if err == nil && reqExt != nil {
+		prebidExt := reqExt.GetPrebid()
+		if prebidExt != nil && prebidExt.Channel != nil {
+			return prebidExt.Channel.Name
+		}
+	}
```

Comment:
Could you handle the case where `err != nil`

Reply:
if case of `err != nil` , catch all is considered for i.e.` channelName = '*' `